data-jdbc/src/test/groovy/io/micronaut/data/jdbc/h2/H2CompositePrimaryKeySpec.groovy

```diff
@@ -72,4 +80,28 @@ class H2CompositePrimaryKeySpec extends Specification {
         then:"The object was deleted"
         project == null
     }
+
+    void "test build findByProjectIdIn"() {
+        given:
+        QueryBuilder encoder = new SqlQueryBuilder()
+        when:
+        def queryModel = QueryModel.from(PersistentEntity.of(Project))
+        def q = encoder.buildQuery(AnnotationMetadata.EMPTY_METADATA, queryModel.eq("projectId", new QueryParameter("projectId")))
+        then:
+        q.query == 'SELECT project_."project_id_department_id",project_."project_id_project_id",LOWER(project_.name) AS name,project_.name AS db_name,UPPER(project_.org) AS org FROM "project" project_ WHERE (project_."project_id_department_id" = ? AND project_."project_id_project_id" = ?)'
+        q.parameters == ["1": "projectId.departmentId", "2": "projectId.projectId"]
+        when:
+        queryModel = QueryModel.from(PersistentEntity.of(Project))
+        q = encoder.buildQuery(AnnotationMetadata.EMPTY_METADATA, queryModel.inList("projectId", new QueryParameter("projectIds")))
+        then:
+        q.query == 'SELECT project_."project_id_department_id",project_."project_id_project_id",LOWER(project_.name) AS name,project_.name AS db_name,UPPER(project_.org) AS org FROM "project" project_ WHERE (project_."project_id_department_id" IN (?) AND project_."project_id_project_id" IN (?))'
```

Comment:
Having two INs is not logically correct. There are two solutions:
1. Running query X times - that how it used to work before, not sure what broke it
2. Some databases support more robust IN that would work for two columns

Reply:
Don't think it ever worked, this issue is from 2020.
Tried approach #2 and that was not doable with how parameter expanding is implemented.